File: base/src/main/java/com/mvp/BasePresenter.java
package com.mvp;

import java.lang.ref.WeakReference;

/**
 * created by zhengweis on 2019/7/25.
 * description：
 */
public class BasePresenter<V extends IBaseView> implements IBasePresenter {
    // 防止 Activity 不走 onDestory() 方法，所以采用弱引用来防止内存泄漏
    private WeakReference<V> mViewRef;

    public BasePresenter(V view) {
        attachView(view);
    }

    public V getView() {
        return mViewRef.get();
    }

    private void attachView(V view) {
        mViewRef = new WeakReference<V>(view);
    }

    @Override
    public boolean isViewAttach() {
        return mViewRef != null && mViewRef.get() != null;
    }

    @Override
    public void detachView() {
        if (mViewRef != null) {
            mViewRef.clear();
            mViewRef = null;
        }
    }
}

File: base/src/main/java/com/mvp/BasePresenter.kt
package com.mvp

import java.lang.ref.WeakReference

/**
 * created by zhengweis on 2019/7/25.
 * description：
 */
open class BasePresenter<V : IBaseView?>(view: V) : IBasePresenter {
    // 防止 Activity 不走 onDestory() 方法，所以采用弱引用来防止内存泄漏
    private var mViewRef: WeakReference<V?>? = null
    val view: V?
        get() = mViewRef!!.get()

    private fun attachView(view: V) {
        mViewRef = WeakReference(view)
    }

    override fun isViewAttach(): Boolean {
        return mViewRef?.get() != null
    }

    override fun detachView() {
        if (mViewRef != null) {
            mViewRef?.clear()
            mViewRef = null
        }
    }

    init {
        attachView(view)
    }
}